beacon-chain/core/blocks/withdrawals_test.go

```diff
@@ -191,3 +196,385 @@ func TestProcessBLSToExecutionChange(t *testing.T) {
+					fullWithdrawal(7, 22), fullWithdrawal(19, 23), fullWithdrawal(28, 24),
+					fullWithdrawal(1, 25),
+				},
+			},
+			Control: control{
+				LastWithdrawalValidatorIndex: 1,
+				NextWithdrawalIndex:          26,
+				Balances:                     map[uint64]uint64{7: 0, 19: 0, 28: 0, 1: 0},
+			},
+		},
+		{
+			Args: args{
+				Name:                         "success many partial withdrawals",
+				NextWithdrawalIndex:          22,
+				LastWithdrawalValidatorIndex: 4,
+				PartialWithdrawalIndices:     []types.ValidatorIndex{7, 19, 28, 1},
+				Withdrawals: []*enginev1.Withdrawal{
+					partialWithdrawal(7, 22), partialWithdrawal(19, 23), partialWithdrawal(28, 24),
+					partialWithdrawal(1, 25),
+				},
+			},
+			Control: control{
+				LastWithdrawalValidatorIndex: 1,
+				NextWithdrawalIndex:          26,
+				Balances: map[uint64]uint64{
+					7:  maxEffectiveBalance,
+					19: maxEffectiveBalance,
+					28: maxEffectiveBalance,
+					1:  maxEffectiveBalance,
+				},
+			},
+		},
+		{
+			Args: args{
+				Name:                         "success many withdrawals",
+				NextWithdrawalIndex:          22,
+				LastWithdrawalValidatorIndex: 12,
+				FullWithdrawalIndices:        []types.ValidatorIndex{7, 19, 28},
+				PartialWithdrawalIndices:     []types.ValidatorIndex{2, 1, 89, 15},
+				Withdrawals: []*enginev1.Withdrawal{
+					partialWithdrawal(15, 22), fullWithdrawal(19, 23), fullWithdrawal(28, 24),
+					partialWithdrawal(89, 25), partialWithdrawal(1, 26), partialWithdrawal(2, 27),
+					fullWithdrawal(7, 28),
+				},
+			},
+			Control: control{
+				LastWithdrawalValidatorIndex: 7,
+				NextWithdrawalIndex:          29,
+				Balances: map[uint64]uint64{
+					7: 0, 19: 0, 28: 0,
+					2: maxEffectiveBalance, 1: maxEffectiveBalance, 89: maxEffectiveBalance,
+					15: maxEffectiveBalance,
+				},
+			},
+		},
+		{
+			Args: args{
+				Name:                         "success more than max fully withdrawals",
+				NextWithdrawalIndex:          22,
+				LastWithdrawalValidatorIndex: 0,
+				FullWithdrawalIndices:        []types.ValidatorIndex{1, 2, 3, 4, 5, 6, 7, 8, 9, 21, 22, 23, 24, 25, 26, 27, 29, 35, 89},
+				Withdrawals: []*enginev1.Withdrawal{
+					fullWithdrawal(1, 22), fullWithdrawal(2, 23), fullWithdrawal(3, 24),
+					fullWithdrawal(4, 25), fullWithdrawal(5, 26), fullWithdrawal(6, 27),
+					fullWithdrawal(7, 28), fullWithdrawal(8, 29), fullWithdrawal(9, 30),
+					fullWithdrawal(21, 31), fullWithdrawal(22, 32), fullWithdrawal(23, 33),
+					fullWithdrawal(24, 34), fullWithdrawal(25, 35), fullWithdrawal(26, 36),
+					fullWithdrawal(27, 37),
+				},
+			},
+			Control: control{
+				LastWithdrawalValidatorIndex: 27,
+				NextWithdrawalIndex:          38,
+				Balances: map[uint64]uint64{
+					1: 0, 2: 0, 3: 0, 4: 0, 5: 0, 6: 0, 7: 0, 8: 0, 9: 0,
+					21: 0, 22: 0, 23: 0, 24: 0, 25: 0, 26: 0, 27: 0,
+				},
+			},
+		},
+		{
+			Args: args{
+				Name:                         "success more than max partially withdrawals",
```

Comment:
We don't have more than max withdrawals here

Reply:
same as above